### DISCUSSION TITLE
Melanoma Detection

### DISCUSSION THREAD

**Discussion by ennriqe** [NillionNetwork/discussions#84] [Show and tell]
### Show and tell project type

Hackathon Project Submission

### Github Repo Link

https://github.com/Lucab95/ETH-global-Melanoma-pred/

### Project Description

Private Classification of Malignant Skin Tumors using a Neural Network

### What problems does your project solve? How does it preserve privacy for users?

- Problem: There are people worried about a skin issue who want a first instant diagnosis before going to the doctor and spending money and time.

Solution: Our tool can provide a preliminary prediction, which by no means replaces a doctor visit but can give some information.

- There are currently apps on the Apple and Play store that one can use for this same purpose, however, they are linked to the user's identity. Those apps could be storing the images to train their models or most importantly, selling the fact that the user has skin cancer or thinks they have skin cancer to different parties such as insurance companies, pharmaceutical companies, healthcare providers, or marketing and advertising firms.

Solution: Our Blind App gives the user a preliminary prediction diagnosis without storing or even directly looking at the image, users can be assured that their consultation is fully private.

### How does the project use Nillion? Describe and link to any Nada programs

We used Nillion to host a Neural Network where the inputs to this model are fully private, in this case the input is a matrix that represents a picture.

Nada program 1: https://github.com/Lucab95/ETH-global-Melanoma-pred/blob/master/quickstart/nada_quickstart_programs/src/multy_percep.py
Nada program 2: https://github.com/Lucab95/ETH-global-Melanoma-pred/blob/master/quickstart/nada_quickstart_programs/src/my_nn.py

### Is there anything else you want to share?

Follow the readme and you will be able to load the web app to try it out yourself, be mindful to include the .env file with your private key as stated, in the same folder as the Steamlit python script.

### Optional - Link your project and team members' social handles

https://x.com/Luca_brglt

https://twitter.com/BarruecoEnrique

### Optional - Team ETH Address(es)

0x4f6E0faec10CF04B428eC9E34cd88AE04Fba6062

0x4e43b435089bF9e7e38E1470E2dF3bA0B560970d

**Comment by manel1874** [comment]
Really cool! Just a note: the github link repo is not complete.

**Reply by ennriqe** [reply]
Thank you!
